packages/api/src/command/medical/patient/converage-assessment-get.ts

```diff
@@ -0,0 +1,143 @@
+import { Patient } from "@metriport/core/domain/patient";
+import { executeAsynchronously } from "@metriport/core/util/concurrency";
+import { createMRSummaryFileName } from "@metriport/core/domain/medical-record-summary";
+import { S3Utils } from "@metriport/core/external/aws/s3";
+import { countResources } from "../../../external/fhir/patient/count-resources";
+import { Config } from "../../../shared/config";
+import { out } from "@metriport/core/util/log";
+import { capture } from "@metriport/core/util/notifications";
+import { errorToString } from "@metriport/shared";
+
+const region = Config.getAWSRegion();
+const bucket = Config.getMedicalDocumentsBucketName();
+
+function getS3UtilsInstance(): S3Utils {
+  return new S3Utils(region);
+}
+
+export type CoverageAssessment = {
+  patientId: string;
+  downloadStatus: string | undefined;
+  docCount: number | undefined;
+  convertStatus: string | undefined;
+  fhirCount: number;
+  fhirDetails: string;
+  mrSummaryUrl: string | undefined;
+};
+
+export type PatientWithCoverageAssessment = Patient &
+  Partial<Omit<CoverageAssessment, "patientId">>;
+
+export async function getCoverageAssessments({
+  cxId,
+  patients,
+}: {
+  cxId: string;
+  patients: Patient[];
+}): Promise<PatientWithCoverageAssessment[]> {
+  const { log } = out(`getCoverageAssessments - cxId ${cxId}`);
+  const patientsWithAssessment: PatientWithCoverageAssessment[] = [];
+
+  async function getCoverageAssessmentWrapper({
+    cxId,
+    patient,
+    patientsWithAssessment,
+    log,
+  }: {
+    cxId: string;
+    patient: Patient;
+    patientsWithAssessment: PatientWithCoverageAssessment[];
+    log: typeof console.log;
+  }): Promise<void> {
+    try {
+      const coverageAssessment = await getCoverageAssessment({ cxId, patient, log });
+      patientsWithAssessment.push({
+        ...patient,
+        ...coverageAssessment,
+      });
+    } catch (error) {
+      const msg = `Failed to get coverage assessment. Cause: ${errorToString(error)}`;
+      capture.error(msg, {
+        extra: {
+          cxId,
+          patientId: patient.id,
+          context: "coverage-assessment.get",
+        },
+      });
```

Comment:
Can we move this outside of the wrapper/promise and send a single message w/ the errors for a given call? The concern is that we would hammer the alerts channel if something is off that leads to all/lots failing.

Reply:
Done.